packages/hash/api/src/graphql/resolvers/entityType/updateEntityType.ts

```diff
@@ -7,13 +9,30 @@ export const updateEntityType: Resolver<
   {},
   LoggedInGraphQLContext,
   MutationUpdateEntityTypeArgs
-> = async (_, { accountId, entityId, schema }, { dataSources, user }) => {
-  const entityType = await EntityType.updateSchema(dataSources.db, {
-    accountId,
-    updatedByAccountId: user.accountId,
-    createdByAccountId: user.entityId,
-    entityId,
-    schema,
+> = async (
+  _,
+  { accountId, entityId, schema },
+  { dataSources: { db }, user },
+) => {
+  const entityType = await EntityType.getEntityType(db, {
+    entityTypeId: entityId,
   });
-  return entityType.toGQLEntityType();
+
+  if (!entityType) {
+    throw new ApolloError(
+      `EntityType with entityId ${entityId} not found`,
+      "NOT_FOUND",
+    );
+  }
+
+  return await db.transaction(async (conn) =>
+    (
+      await entityType.update(conn, {
+        accountId,
+        updatedByAccountId: user.accountId,
+        createdByAccountId: user.entityId,
+        schema,
+      })
+    ).toGQLEntityType(),
+  );
```

Comment:
I think the following would be more readable:
```suggestion
  return await db.transaction(async (conn) => {
    await entityType.update(conn, {
      accountId,
      updatedByAccountId: user.accountId,
      createdByAccountId: user.entityId,
      schema,
    });

    return entityType.toGQLEntityType();
  });
```

Reply:
(I now realise you were doing this to make use of the return type of the `update` method, but once [this](https://github.com/hashintel/hash/pull/369#discussion_r820528519) has been addressed you'll no longer have to do that)

Reply:
Addressed along with having transaction around all of resolver :+1: